satpy/tests/reader_tests/test_hsaf_nc.py

```diff
@@ -0,0 +1,186 @@
+import pytest
+import xarray as xr
+from pyresample import AreaDefinition
+
+from satpy._config import config_search_paths
+from satpy.readers.core.loading import load_reader
+
+# the readers file type
+FILE_TYPE_H60 = "hsaf_h60_nc"
+FILE_TYPE_H63 = "hsaf_h63_nc"
+FILE_TYPE_H90 = "hsaf_h90_nc"
+
+# parameters per file type
+FILE_PARAMS = {
+    FILE_TYPE_H60: {
+        "fake_file": "h60_20251105_0000_fdk.nc",
+        "real_file": "/".join(os.path.abspath(__file__).split("/")[0:-1]) + "/data/h60_20251111_1100_fdk.nc.gz",
+        "yaml_file": "hsaf_nc.yaml",
+    },
+    FILE_TYPE_H63: {
+        "fake_file": "h63_20251105_0000_fdk.nc",
+        "real_file": "/".join(os.path.abspath(__file__).split("/")[0:-1]) + "/data/h63_20251111_1100_fdk.nc.gz",
+        "yaml_file": "hsaf_nc.yaml",
+    },
+    FILE_TYPE_H90: {
+        "fake_file": "h90_20251105_0000_fdk.nc",
+        "real_file": "/".join(os.path.abspath(__file__).split("/")[0:-1]) + "/data/h90_20251111_1000_01_fdk.nc.gz",
+        "yaml_file": "hsaf_nc.yaml",
+    }
+}
+
+# Avoid too many arguments for test_load_datasets
+LoadDatasetsParams = namedtuple(
+    "LoadDatasetsParams",
+    ["file_type", "loadable_ids", "unit", "resolution", "area_name"]
+)
+
+# constants for fake test data
+DEFAULT_SHAPE = (5, 5)
+rng = np.random.default_rng()
+DEFAULT_RR = rng.random(DEFAULT_SHAPE)
+DEFAULT_QIND = rng.integers(0, 100, size=DEFAULT_SHAPE)
+
+def fake_hsaf_dataset(filename, **kwargs):
+    """Mimic a HSAF NetCDF file content."""
+    ds = xr.Dataset(
+        {
+            "rr": (("ny", "nx"), DEFAULT_RR),
+            "acc_rr": (("ny", "nx"), DEFAULT_RR),
+            "qind": (("ny", "nx"), DEFAULT_QIND),
+        },
+        coords={"ny": np.arange(DEFAULT_SHAPE[0]),
+                "nx": np.arange(DEFAULT_SHAPE[1])},
+        attrs={"satellite_identifier": "MSG1",
+               "start_time": "2025-11-05T00:00:00",
+               "sub_satellite_longitude": "0.0f"}
+    )
+    return ds
+
+class TestHSAFNCReader:
+    """Test HSAF H60 NetCDF reader."""
+
+    def setup_method(self):
+        """Load reader YAML and configs."""
+        for file_type, params in FILE_PARAMS.items():
+            # search for the yaml config
+            params["reader_configs"] = config_search_paths(os.path.join("readers", params["yaml_file"]))
+            # load the reader
+            params["reader"] = load_reader(params["reader_configs"], name = file_type)
+
+    @pytest.mark.parametrize(
+        ("file_type", "expected_loadables"),
+        [
+            (FILE_PARAMS[FILE_TYPE_H60], 1),
+            (FILE_PARAMS[FILE_TYPE_H63], 1),
+            (FILE_PARAMS[FILE_TYPE_H90], 1),
+        ],
+    )
+    def test_reader_creation(self, file_type, expected_loadables):
+        """Test that the reader can create file handlers."""
+        with mock.patch("satpy.readers.hsaf_nc.generic_open") as mock_generic_open, \
+                mock.patch("satpy.readers.hsaf_nc.xr.open_dataset") as mock_open_dataset:
```

Comment:
We have now as policy to try to create a stub file in `tmp_path` instead of mocking. I think here, it's a simple matter of writing the xarray dataset to disk and use that as input file. With the added benefit that you can (g/b)zip it to test the decompression...